src/librustdoc/html/highlight.rs

```diff
@@ -226,6 +240,43 @@ impl<F: Write> Drop for TokenHandler<'_, '_, F> {
     }
 }
 
+fn write_scraped_line_number(out: &mut impl Write, line: u32, extra: &'static str) {
+    // https://developers.google.com/search/docs/crawling-indexing/robots-meta-tag#data-nosnippet-attr
+    // Do not show "1 2 3 4 5 ..." in web search results.
+    write!(out, "{extra}<span data-nosnippet>{line}</span>",).unwrap();
+}
+
+fn write_line_number(out: &mut impl Write, line: u32, extra: &'static str) {
+    // https://developers.google.com/search/docs/crawling-indexing/robots-meta-tag#data-nosnippet-attr
+    // Do not show "1 2 3 4 5 ..." in web search results.
+    write!(out, "{extra}<a href=\"#{line}\" id={line} data-nosnippet>{line}</a>",).unwrap();
```

Comment:
... I think?

```suggestion
    write!(out, "{extra}<a href=\"#{line}\" id=\"{line}\" data-nosnippet>{line}</a>",).unwrap();
```

Reply:
It's a number so I'm cheating here. That allows to get 2 characters x number of lines, so it becomes interesting for big crates.

Reply:
Cool, didn't know that was valid in HTML5!

Reply:
It's valid in all HTML as far as I know. ^^'